src/sage/schemes/weighted_projective/weighted_projective_point.py

```diff
@@ -0,0 +1,297 @@
+                raise TypeError(f"ambient space {X} must be a weighted projective space")
+
+            from sage.rings.ring import CommutativeRing
+            d = X.codomain().ambient_space().ngens()
+            if isinstance(v, SchemeMorphism):
+                v = list(v)
+            else:
+                try:
+                    if isinstance(v.parent(), CommutativeRing):
+                        v = [v]
+                except AttributeError:
+                    pass
+            if not isinstance(v, (list, tuple)):
+                raise TypeError("argument v (= %s) must be a scheme point, list, or tuple" % str(v))
+            if len(v) != d and len(v) != d-1:
+                raise TypeError("v (=%s) must have %s components" % (v, d))
+
+            R = X.value_ring()
+            v = Sequence(v, R)
+            if len(v) == d-1:     # very common special case
+                v.append(R.one())
+
+            # (0 : 0 : ... : 0) is not a valid (weighted) projective point
+            if not any(v):
+                raise ValueError(f"{v} does not define a valid projective "
+                                 "point since all entries are zero")
+
+            # over other rings, we do not have a generic method to check
+            # whether the given coordinates is a multiple of a zero divisor
+            # so we just let it pass.
+            X.extended_codomain()._check_satisfies_equations(v)
+
+            self._coords = tuple(v)
+
+            # we (try to) normalise coordinates!
+            self.normalize_coordinates()
+
+        else:
+            self._coords = tuple(v)
+
+    def _repr_(self):
+        return "({})".format(" : ".join(map(repr, self._coords)))
+
+    def _richcmp_(self, other, op):
+        """
+        Test the weighted projective equality of two points.
+
+        INPUT:
+
+        - ``other`` -- a point on weighted projective space
+
+        OUTPUT:
+
+        Boolean
+
+        EXAMPLES::
+
+            sage: WP = WeightedProjectiveSpace([3, 4, 5], QQ)
+            sage: u = abs(QQ.random_element()) + 1 / 4 # nonzero
+            sage: P, Q = WP(2, 3, 4), WP(2 * u^3, 3 * u^4, 4 * u^5)
+            sage: P == Q
+            True
+            sage: P != Q
+            False
+
+        ::
+
+            sage: P, Q = WP(2, 3, 4), WP(2, 3, 5)
+            sage: P < Q
+            True
+            sage: P <= Q
+            True
+            sage: P > Q
+            False
+            sage: P >= Q
+            False
+        """
+        assert isinstance(other, SchemeMorphism_point)
+
+        space = self.codomain()
+        if space is not other.codomain():
+            return op == op_NE
+
+        if op in [op_EQ, op_NE]:
+            weights = space._weights
+            # (other[i] / self[i])^(1 / weight[i]) all equal
+            prod_weights = prod(weights)
+            # check weights
+            bw = weights == other.codomain()._weights
+            if bw != (op == op_EQ):
```

Comment:
```suggestion
            if (weights == other.codomain()._weights) != (op == op_EQ):
```
Unless this exceeds the line length limit, let's remove the unnecessary temporary variable.